apps/api/v2/src/ee/event-types/event-types_2024_04_15/services/event-types.service.ts

```diff
@@ -127,8 +128,14 @@ export class EventTypesService_2024_04_15 {
   async updateEventType(eventTypeId: number, body: UpdateEventTypeInput_2024_04_15, user: UserWithProfile) {
     this.checkCanUpdateEventType(user.id, eventTypeId);
     const eventTypeUser = await this.getUserToUpdateEvent(user);
+    const bookingFields = [...(body.bookingFields || [])];
+
+    if (!bookingFields.find((field) => field.type === "email")) {
+      bookingFields.push(systemBeforeFieldEmail);
+    }
```

Comment:
should it check for both email and phone number? and push email only if one is missing ?

Reply:
Good call - implemented in [refactor: push email in case of missing phone](https://github.com/calcom/cal.com/pull/16685/commits/3c68ffcda7defd0071ff8f097e00aacf40277613)